### PR TITLE
[Sol->Yul] Fix bare call wihout returndatacopy.

### LINKED ISSUE #10919: [Sol->Yul] Invalid code generation due to address.call() for spuriousDragon only
## Description

```
contract C {
function f() public returns (bool success) {
    (success, ) = (address(0)).call{value: 30}("");
}
}
```

results in invalid Yul code gen for spuriousDragon (```$ solc --ir --evm-version spuriousDragon test.sol```)

```
Internal compiler error during compilation:
/solidity/libsolidity/codegen/ir/IRGenerator.cpp(64): Throw in function pair<std::string, std::string> solidity::frontend::IRGenerator::run(const solidity::frontend::ContractDefinition &, const map<const solidity::frontend::ContractDefinition *, const std::string_view> &)
Dynamic exception type: boost::wrapexcept<solidity::langutil::InternalCompilerError>
std::exception::what: 
object "C_21" {
    code {
        mstore(64, 128)
        if callvalue() { revert(0, 0) }

        constructor_C_21()

        codecopy(0, dataoffset("C_21_deployed"), datasize("C_21_deployed"))

        return(0, datasize("C_21_deployed"))

        function constructor_C_21() {

        }

    }
    object "C_21_deployed" {
        code {
            mstore(64, 128)

            if iszero(lt(calldatasize(), 4))
            {
                let selector := shift_right_224_unsigned(calldataload(0))
                switch selector

                case 0x26121ff0
                {
                    // f()

                    if callvalue() { revert(0, 0) }
                    abi_decode_tuple_(4, calldatasize())
                    let ret_0 :=  fun_f_20()
                    let memPos := allocate_memory(0)
                    let memEnd := abi_encode_tuple_t_bool__to_t_bool__fromStack(memPos , ret_0)
                    return(memPos, sub(memEnd, memPos))
                }

                default {}
            }
            if iszero(calldatasize()) {  }
            revert(0, 0)

            function abi_decode_tuple_(headStart, dataEnd)   {
                if slt(sub(dataEnd, headStart), 0) { revert(0, 0) }

            }

            function abi_encode_t_bool_to_t_bool_fromStack(value, pos) {
                mstore(pos, cleanup_t_bool(value))
            }

            function abi_encode_t_stringliteral_c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470_to_t_bytes_memory_ptr_nonPadded_inplace_fromStack(pos) -> end {
                pos := array_storeLengthForEncoding_t_bytes_memory_ptr_nonPadded_inplace_fromStack(pos, 0)
                store_literal_in_memory_c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470(pos)
                end := add(pos, 0)
            }

            function abi_encode_tuple_packed_t_stringliteral_c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470__to_t_bytes_memory_ptr__nonPadded_inplace_fromStack(pos ) -> end {

                pos := abi_encode_t_stringliteral_c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470_to_t_bytes_memory_ptr_nonPadded_inplace_fromStack( pos)

                end := pos
            }

            function abi_encode_tuple_t_bool__to_t_bool__fromStack(headStart , value0) -> tail {
                tail := add(headStart, 32)

                abi_encode_t_bool_to_t_bool_fromStack(value0,  add(headStart, 0))

            }

            function allocate_memory(size) -> memPtr {
                memPtr := allocate_unbounded()
                finalize_allocation(memPtr, size)
            }

            function allocate_unbounded() -> memPtr {
                memPtr := mload(64)
            }

            function array_storeLengthForEncoding_t_bytes_memory_ptr_nonPadded_inplace_fromStack(pos, length) -> updated_pos {
                updated_pos := pos
            }

            function cleanup_t_bool(value) -> cleaned {
                cleaned := iszero(iszero(value))
            }

            function cleanup_t_uint160(value) -> cleaned {
                cleaned := and(value, 0xffffffffffffffffffffffffffffffffffffffff)
            }

            function cleanup_t_uint256(value) -> cleaned {
                cleaned := value
            }

            function convert_t_bool_to_t_bool(value) -> converted {
                converted := cleanup_t_bool(value)
            }

            function convert_t_rational_0_by_1_to_t_address(value) -> converted {
                converted := convert_t_rational_0_by_1_to_t_uint160(value)
            }

            function convert_t_rational_0_by_1_to_t_uint160(value) -> converted {
                converted := cleanup_t_uint160(value)
            }

            function convert_t_rational_30_by_1_to_t_uint256(value) -> converted {
                converted := cleanup_t_uint256(value)
            }

            function convert_t_tuple$_t_bool_$_t_inaccessible_$_to_t_tuple$_t_bool_$__$(value0, value1) -> converted0 {
                converted0 := convert_t_bool_to_t_bool(value0)

            }

            function finalize_allocation(memPtr, size) {
                let newFreePtr := add(memPtr, round_up_to_mul_of_32(size))
                // protect against overflow
                if or(gt(newFreePtr, 0xffffffffffffffff), lt(newFreePtr, memPtr)) { panic_error_0x41() }
                mstore(64, newFreePtr)
            }

            function fun_f_20() -> vloc_success_3 {
                let zero_value_for_type_t_bool_1 := zero_value_for_split_t_bool()
                vloc_success_3 := zero_value_for_type_t_bool_1

                let expr_9 := 0x00
                let expr_10 := convert_t_rational_0_by_1_to_t_address(expr_9)
                let expr_11 := expr_10
                let expr_12_address := expr_11
                let expr_13 := 0x1e
                let expr_14_address := expr_12_address
                let expr_14_value := convert_t_rational_30_by_1_to_t_uint256(expr_13)

                let _2 := allocate_unbounded()
                let _3 := sub(abi_encode_tuple_packed_t_stringliteral_c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470__to_t_bytes_memory_ptr__nonPadded_inplace_fromStack(_2  ), _2)

                let expr_16_component_1 := call(gas(), expr_14_address,  expr_14_value,  _2, _3, 0, 0)

                let _4_component_1 := convert_t_tuple$_t_bool_$_t_inaccessible_$_to_t_tuple$_t_bool_$__$(expr_16_component_1, expr_16_component_2)
                vloc_success_3 := _4_component_1

            }

            function panic_error_0x41() {
                mstore(0, 35408467139433450592217433187231851964531694900788300625387963629091585785856)
                mstore(4, 0x41)
                revert(0, 0x24)
            }

            function round_up_to_mul_of_32(value) -> result {
                result := and(add(value, 31), not(31))
            }

            function shift_right_224_unsigned(value) -> newValue {
                newValue :=

                div(value, 0x0100000000000000000000000000000000000000000000000000000000)

            }

            function store_literal_in_memory_c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470(memPtr) {

            }

            function zero_value_for_split_t_bool() -> ret {
                ret := 0
            }

        }

    }

}



Invalid IR generated:
Error: Identifier not found.



[solidity::util::tag_comment*] = 
object "C_21" {
    code {
        mstore(64, 128)
        if callvalue() { revert(0, 0) }

        constructor_C_21()

        codecopy(0, dataoffset("C_21_deployed"), datasize("C_21_deployed"))

        return(0, datasize("C_21_deployed"))

        function constructor_C_21() {

        }

    }
    object "C_21_deployed" {
        code {
            mstore(64, 128)

            if iszero(lt(calldatasize(), 4))
            {
                let selector := shift_right_224_unsigned(calldataload(0))
                switch selector

                case 0x26121ff0
                {
                    // f()

                    if callvalue() { revert(0, 0) }
                    abi_decode_tuple_(4, calldatasize())
                    let ret_0 :=  fun_f_20()
                    let memPos := allocate_memory(0)
                    let memEnd := abi_encode_tuple_t_bool__to_t_bool__fromStack(memPos , ret_0)
                    return(memPos, sub(memEnd, memPos))
                }

                default {}
            }
            if iszero(calldatasize()) {  }
            revert(0, 0)

            function abi_decode_tuple_(headStart, dataEnd)   {
                if slt(sub(dataEnd, headStart), 0) { revert(0, 0) }

            }

            function abi_encode_t_bool_to_t_bool_fromStack(value, pos) {
                mstore(pos, cleanup_t_bool(value))
            }

            function abi_encode_t_stringliteral_c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470_to_t_bytes_memory_ptr_nonPadded_inplace_fromStack(pos) -> end {
                pos := array_storeLengthForEncoding_t_bytes_memory_ptr_nonPadded_inplace_fromStack(pos, 0)
                store_literal_in_memory_c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470(pos)
                end := add(pos, 0)
            }

            function abi_encode_tuple_packed_t_stringliteral_c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470__to_t_bytes_memory_ptr__nonPadded_inplace_fromStack(pos ) -> end {

                pos := abi_encode_t_stringliteral_c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470_to_t_bytes_memory_ptr_nonPadded_inplace_fromStack( pos)

                end := pos
            }

            function abi_encode_tuple_t_bool__to_t_bool__fromStack(headStart , value0) -> tail {
                tail := add(headStart, 32)

                abi_encode_t_bool_to_t_bool_fromStack(value0,  add(headStart, 0))

            }

            function allocate_memory(size) -> memPtr {
                memPtr := allocate_unbounded()
                finalize_allocation(memPtr, size)
            }

            function allocate_unbounded() -> memPtr {
                memPtr := mload(64)
            }

            function array_storeLengthForEncoding_t_bytes_memory_ptr_nonPadded_inplace_fromStack(pos, length) -> updated_pos {
                updated_pos := pos
            }

            function cleanup_t_bool(value) -> cleaned {
                cleaned := iszero(iszero(value))
            }

            function cleanup_t_uint160(value) -> cleaned {
                cleaned := and(value, 0xffffffffffffffffffffffffffffffffffffffff)
            }

            function cleanup_t_uint256(value) -> cleaned {
                cleaned := value
            }

            function convert_t_bool_to_t_bool(value) -> converted {
                converted := cleanup_t_bool(value)
            }

            function convert_t_rational_0_by_1_to_t_address(value) -> converted {
                converted := convert_t_rational_0_by_1_to_t_uint160(value)
            }

            function convert_t_rational_0_by_1_to_t_uint160(value) -> converted {
                converted := cleanup_t_uint160(value)
            }

            function convert_t_rational_30_by_1_to_t_uint256(value) -> converted {
                converted := cleanup_t_uint256(value)
            }

            function convert_t_tuple$_t_bool_$_t_inaccessible_$_to_t_tuple$_t_bool_$__$(value0, value1) -> converted0 {
                converted0 := convert_t_bool_to_t_bool(value0)

            }

            function finalize_allocation(memPtr, size) {
                let newFreePtr := add(memPtr, round_up_to_mul_of_32(size))
                // protect against overflow
                if or(gt(newFreePtr, 0xffffffffffffffff), lt(newFreePtr, memPtr)) { panic_error_0x41() }
                mstore(64, newFreePtr)
            }

            function fun_f_20() -> vloc_success_3 {
                let zero_value_for_type_t_bool_1 := zero_value_for_split_t_bool()
                vloc_success_3 := zero_value_for_type_t_bool_1

                let expr_9 := 0x00
                let expr_10 := convert_t_rational_0_by_1_to_t_address(expr_9)
                let expr_11 := expr_10
                let expr_12_address := expr_11
                let expr_13 := 0x1e
                let expr_14_address := expr_12_address
                let expr_14_value := convert_t_rational_30_by_1_to_t_uint256(expr_13)

                let _2 := allocate_unbounded()
                let _3 := sub(abi_encode_tuple_packed_t_stringliteral_c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470__to_t_bytes_memory_ptr__nonPadded_inplace_fromStack(_2  ), _2)

                let expr_16_component_1 := call(gas(), expr_14_address,  expr_14_value,  _2, _3, 0, 0)

                let _4_component_1 := convert_t_tuple$_t_bool_$_t_inaccessible_$_to_t_tuple$_t_bool_$__$(expr_16_component_1, expr_16_component_2)
                vloc_success_3 := _4_component_1

            }

            function panic_error_0x41() {
                mstore(0, 35408467139433450592217433187231851964531694900788300625387963629091585785856)
                mstore(4, 0x41)
                revert(0, 0x24)
            }

            function round_up_to_mul_of_32(value) -> result {
                result := and(add(value, 31), not(31))
            }

            function shift_right_224_unsigned(value) -> newValue {
                newValue :=

                div(value, 0x0100000000000000000000000000000000000000000000000000000000)

            }

            function store_literal_in_memory_c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470(memPtr) {

            }

            function zero_value_for_split_t_bool() -> ret {
                ret := 0
            }

        }

    }

}



Invalid IR generated:
Error: Identifier not found.
```

### LINKED ISSUE #10920: [Sol->Yul] Invalid code generation for address call against spuriousDragon and homstead
## Description

```
contract C {
        function f() public { (address(1).call("")); }
}
```

results in invalid Yul code (```solc --ir --evm-version homestead test.sol``` likewise spuriousDragon)

```
Internal compiler error during compilation:
/solidity/libsolidity/codegen/ir/IRGenerator.cpp(64): Throw in function pair<std::string, std::string> solidity::frontend::IRGenerator::run(const solidity::frontend::ContractDefinition &, const map<const solidity::frontend::ContractDefinition *, const std::string_view> &)
Dynamic exception type: boost::wrapexcept<solidity::langutil::InternalCompilerError>
std::exception::what: 
object "C_14" {
    code {
        mstore(64, 128)
        if callvalue() { revert(0, 0) }

        constructor_C_14()

        codecopy(0, dataoffset("C_14_deployed"), datasize("C_14_deployed"))

        return(0, datasize("C_14_deployed"))

        function constructor_C_14() {

        }

    }
    object "C_14_deployed" {
        code {
            mstore(64, 128)

            if iszero(lt(calldatasize(), 4))
            {
                let selector := shift_right_224_unsigned(calldataload(0))
                switch selector

                case 0x26121ff0
                {
                    // f()

                    if callvalue() { revert(0, 0) }
                    abi_decode_tuple_(4, calldatasize())
                    fun_f_13()
                    let memPos := allocate_memory(0)
                    let memEnd := abi_encode_tuple__to__fromStack(memPos  )
                    return(memPos, sub(memEnd, memPos))
                }

                default {}
            }
            if iszero(calldatasize()) {  }
            revert(0, 0)

            function abi_decode_tuple_(headStart, dataEnd)   {
                if slt(sub(dataEnd, headStart), 0) { revert(0, 0) }

            }

            function abi_encode_t_stringliteral_c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470_to_t_bytes_memory_ptr_nonPadded_inplace_fromStack(pos) -> end {
                pos := array_storeLengthForEncoding_t_bytes_memory_ptr_nonPadded_inplace_fromStack(pos, 0)
                store_literal_in_memory_c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470(pos)
                end := add(pos, 0)
            }

            function abi_encode_tuple__to__fromStack(headStart ) -> tail {
                tail := add(headStart, 0)

            }

            function abi_encode_tuple_packed_t_stringliteral_c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470__to_t_bytes_memory_ptr__nonPadded_inplace_fromStack(pos ) -> end {

                pos := abi_encode_t_stringliteral_c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470_to_t_bytes_memory_ptr_nonPadded_inplace_fromStack( pos)

                end := pos
            }

            function allocate_memory(size) -> memPtr {
                memPtr := allocate_unbounded()
                finalize_allocation(memPtr, size)
            }

            function allocate_unbounded() -> memPtr {
                memPtr := mload(64)
            }

            function array_storeLengthForEncoding_t_bytes_memory_ptr_nonPadded_inplace_fromStack(pos, length) -> updated_pos {
                updated_pos := pos
            }

            function cleanup_t_uint160(value) -> cleaned {
                cleaned := and(value, 0xffffffffffffffffffffffffffffffffffffffff)
            }

            function convert_t_rational_1_by_1_to_t_address(value) -> converted {
                converted := convert_t_rational_1_by_1_to_t_uint160(value)
            }

            function convert_t_rational_1_by_1_to_t_uint160(value) -> converted {
                converted := cleanup_t_uint160(value)
            }

            function finalize_allocation(memPtr, size) {
                let newFreePtr := add(memPtr, round_up_to_mul_of_32(size))
                // protect against overflow
                if or(gt(newFreePtr, 0xffffffffffffffff), lt(newFreePtr, memPtr)) { panic_error_0x41() }
                mstore(64, newFreePtr)
            }

            function fun_f_13() {

                let expr_5 := 0x01
                let expr_6 := convert_t_rational_1_by_1_to_t_address(expr_5)
                let expr_7_address := expr_6

                let _1 := allocate_unbounded()
                let _2 := sub(abi_encode_tuple_packed_t_stringliteral_c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470__to_t_bytes_memory_ptr__nonPadded_inplace_fromStack(_1  ), _1)

                let expr_9_component_1 := call(gas(), expr_7_address,  0,  _1, _2, 0, 0)

                let expr_10_component_1 := expr_9_component_1
                let expr_10_component_2 := expr_9_component_2

            }

            function panic_error_0x41() {
                mstore(0, 35408467139433450592217433187231851964531694900788300625387963629091585785856)
                mstore(4, 0x41)
                revert(0, 0x24)
            }

            function round_up_to_mul_of_32(value) -> result {
                result := and(add(value, 31), not(31))
            }

            function shift_right_224_unsigned(value) -> newValue {
                newValue :=

                div(value, 0x0100000000000000000000000000000000000000000000000000000000)

            }

            function store_literal_in_memory_c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470(memPtr) {

            }

        }

    }

}



Invalid IR generated:
Error: Identifier not found.



[solidity::util::tag_comment*] = 
object "C_14" {
    code {
        mstore(64, 128)
        if callvalue() { revert(0, 0) }

        constructor_C_14()

        codecopy(0, dataoffset("C_14_deployed"), datasize("C_14_deployed"))

        return(0, datasize("C_14_deployed"))

        function constructor_C_14() {

        }

    }
    object "C_14_deployed" {
        code {
            mstore(64, 128)

            if iszero(lt(calldatasize(), 4))
            {
                let selector := shift_right_224_unsigned(calldataload(0))
                switch selector

                case 0x26121ff0
                {
                    // f()

                    if callvalue() { revert(0, 0) }
                    abi_decode_tuple_(4, calldatasize())
                    fun_f_13()
                    let memPos := allocate_memory(0)
                    let memEnd := abi_encode_tuple__to__fromStack(memPos  )
                    return(memPos, sub(memEnd, memPos))
                }

                default {}
            }
            if iszero(calldatasize()) {  }
            revert(0, 0)

            function abi_decode_tuple_(headStart, dataEnd)   {
                if slt(sub(dataEnd, headStart), 0) { revert(0, 0) }

            }

            function abi_encode_t_stringliteral_c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470_to_t_bytes_memory_ptr_nonPadded_inplace_fromStack(pos) -> end {
                pos := array_storeLengthForEncoding_t_bytes_memory_ptr_nonPadded_inplace_fromStack(pos, 0)
                store_literal_in_memory_c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470(pos)
                end := add(pos, 0)
            }

            function abi_encode_tuple__to__fromStack(headStart ) -> tail {
                tail := add(headStart, 0)

            }

            function abi_encode_tuple_packed_t_stringliteral_c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470__to_t_bytes_memory_ptr__nonPadded_inplace_fromStack(pos ) -> end {

                pos := abi_encode_t_stringliteral_c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470_to_t_bytes_memory_ptr_nonPadded_inplace_fromStack( pos)

                end := pos
            }

            function allocate_memory(size) -> memPtr {
                memPtr := allocate_unbounded()
                finalize_allocation(memPtr, size)
            }

            function allocate_unbounded() -> memPtr {
                memPtr := mload(64)
            }

            function array_storeLengthForEncoding_t_bytes_memory_ptr_nonPadded_inplace_fromStack(pos, length) -> updated_pos {
                updated_pos := pos
            }

            function cleanup_t_uint160(value) -> cleaned {
                cleaned := and(value, 0xffffffffffffffffffffffffffffffffffffffff)
            }

            function convert_t_rational_1_by_1_to_t_address(value) -> converted {
                converted := convert_t_rational_1_by_1_to_t_uint160(value)
            }

            function convert_t_rational_1_by_1_to_t_uint160(value) -> converted {
                converted := cleanup_t_uint160(value)
            }

            function finalize_allocation(memPtr, size) {
                let newFreePtr := add(memPtr, round_up_to_mul_of_32(size))
                // protect against overflow
                if or(gt(newFreePtr, 0xffffffffffffffff), lt(newFreePtr, memPtr)) { panic_error_0x41() }
                mstore(64, newFreePtr)
            }

            function fun_f_13() {

                let expr_5 := 0x01
                let expr_6 := convert_t_rational_1_by_1_to_t_address(expr_5)
                let expr_7_address := expr_6

                let _1 := allocate_unbounded()
                let _2 := sub(abi_encode_tuple_packed_t_stringliteral_c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470__to_t_bytes_memory_ptr__nonPadded_inplace_fromStack(_1  ), _1)

                let expr_9_component_1 := call(gas(), expr_7_address,  0,  _1, _2, 0, 0)

                let expr_10_component_1 := expr_9_component_1
                let expr_10_component_2 := expr_9_component_2

            }

            function panic_error_0x41() {
                mstore(0, 35408467139433450592217433187231851964531694900788300625387963629091585785856)
                mstore(4, 0x41)
                revert(0, 0x24)
            }

            function round_up_to_mul_of_32(value) -> result {
                result := and(add(value, 31), not(31))
            }

            function shift_right_224_unsigned(value) -> newValue {
                newValue :=

                div(value, 0x0100000000000000000000000000000000000000000000000000000000)

            }

            function store_literal_in_memory_c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470(memPtr) {

            }

        }

    }

}



Invalid IR generated:
Error: Identifier not found.
```

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/ethereum/solidity/issues/10920
Fixes https://github.com/ethereum/solidity/issues/10919

### CODE REVIEW

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp [hunk 1]
```diff
@@ -2542,13 +2540,8 @@ void IRGeneratorForStatements::appendBareCall(
...
 	templ("success", IRVariable(_functionCall).tupleComponent(0).name());
-	if (IRVariable(_functionCall).tupleComponent(1).type().category() == Type::Category::InaccessibleDynamic)
-		templ("returndataVar", "");
-	else
-	{
-		templ("returndataVar", IRVariable(_functionCall).tupleComponent(1).part("mpos").name());
-		templ("extractReturndataFunction", m_utils.extractReturndataFunction());
-	}
+	templ("returndataVar", IRVariable(_functionCall).tupleComponent(1).commaSeparatedList());
+	templ("extractReturndataFunction", m_utils.extractReturndataFunction());
```
Comment (chriseth) [inline libsolidity/codegen/ir/IRGeneratorForStatements.cpp:2544]: The `extractReturndataFunction` already has a special case if there is no `returndatacopy`: It returns the empty byte array.

Comment (bshastry) [reply libsolidity/codegen/ir/IRGeneratorForStatements.cpp:2544]: Note to self: For EVM versions that do not support return data e.g., spurious dragon, empty byte array is returned.

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp [hunk 2]
```diff
@@ -2521,9 +2521,7 @@ void IRGeneratorForStatements::appendBareCall(
 		</needsEncoding>
 
 		let <success> := <call>(<gas>, <address>, <?+value> <value>, </+value> <pos>, <length>, 0, 0)
-		<?+returndataVar>
-			let <returndataVar> := <extractReturndataFunction>()
-		</+returndataVar>
+		let <returndataVar> := <extractReturndataFunction>()
```
Comment (bshastry) [inline libsolidity/codegen/ir/IRGeneratorForStatements.cpp:2524]: This check is now not done by the caller. Check is always done in callee i.e., extractReturndataFunction

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp (hunk 1 above)
Comment (bshastry) [reply libsolidity/codegen/ir/IRGeneratorForStatements.cpp:2544]: @chriseth So for evm versions that did not have return data, how did the caller figure out if a bare call is a success or not?

**Review by bshastry — COMMENTED** [review]
Do Sol to Yul bug fixes begin to feature in Changelog? Or do we do that once the new code gen has been tested for some more time?

**Comment by chriseth** [comment]
@bshastry success is still a flag on the stack

**Comment by chriseth** [comment]
No changelog entry needed.

**Review by cameel — APPROVED** [review]
LGTM

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp [hunk 3]
```diff
@@ -2542,13 +2540,8 @@ void IRGeneratorForStatements::appendBareCall(
 	}
 
 	templ("success", IRVariable(_functionCall).tupleComponent(0).name());
-	if (IRVariable(_functionCall).tupleComponent(1).type().category() == Type::Category::InaccessibleDynamic)
-		templ("returndataVar", "");
-	else
-	{
-		templ("returndataVar", IRVariable(_functionCall).tupleComponent(1).part("mpos").name());
-		templ("extractReturndataFunction", m_utils.extractReturndataFunction());
-	}
```
Comment (bshastry) [inline libsolidity/codegen/ir/IRGeneratorForStatements.cpp:2551]: Why has the distinction between inaccessiblydynamic return types been removed?

Comment (chriseth) [reply libsolidity/codegen/ir/IRGeneratorForStatements.cpp:2551]: Because `m_utils.extractReturndataFunction` already handles both cases.

**Pull request merged**